src/main.cpp

```diff
@@ -1611,12 +1614,12 @@ void Misbehaving(NodeId pnode, int howmuch)
 
     state->nMisbehavior += howmuch;
     int banscore = GetArg("-banscore", DEFAULT_BANSCORE_THRESHOLD);
-    if (state->nMisbehavior >= banscore && state->nMisbehavior - howmuch < banscore)
+    if (state->nMisbehavior >= banscore && !state->fShouldBan)
     {
-        LogPrintf("%s: %s (%d -> %d) BAN THRESHOLD EXCEEDED\n", __func__, state->name, state->nMisbehavior-howmuch, state->nMisbehavior);
```

Comment:
No need to add id to state, as CNodeState is exactly indexed by NodeId already (meaning you won't ever have a CNodeState without knowing its NodeId); specifically, pnode here contains the NodeId.